```
[100%] Building yosys

  Build successful.

'./yosys-abc' -> '/nix/store/fz6zvx79q9lgdzx0p820db1d7lqjphrg-abc-verifier-unstable-2023-02-23/bin/abc'
buildPhase completed in 3 minutes 48 seconds
running tests
check flags: -j8 SHELL=/nix/store/f2fi9x1fnwly702yqdlsjhy46pcgz5vi-bash-5.2-p15/bin/bash PREFIX=/nix/store/3ylzzagg57za030ss0q312i344h6y12c-yosys-0.30 VERBOSE=y test
[Makefile.conf] CONFIG := clang
[Makefile.conf] ABCEXTERNAL = /nix/store/fz6zvx79q9lgdzx0p820db1d7lqjphrg-abc-verifier-unstable-2023-02-23/bin/abc
cd tests/simple && bash run-test.sh ""
make[1]: Entering directory '/private/tmp/nix-build-yosys-0.30.drv-0/source/tests/simple'
+ clang -Wall -o /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata.c
+ clang -Wall -o /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata.c
+ clang -Wall -o /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata.c
+ clang -Wall -o /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata.c
+ clang -Wall -o /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata.c
+ clang -Wall -o /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata.c
+ clang -Wall -o /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata.c
+ clang -Wall -o /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata.c
libc++abi: terminating with uncaught exception of type std::runtime_error: opening input file: No such file or directory
/nix/store/065rv9vqknm20xqrylz7fmmvq6azsagi-post-link-sign-hook: line 4: 57848 Abort trap: 6           CODESIGN_ALLOCATE=codesign_allocate /nix/store/37yhd1z0xw2rpdy5g4k526h1a87i8qm0-sigtool-0.1.3/bin/codesign -f -s - "$linkerOutput"
libc++abi: terminating with uncaught exception of type std::runtime_error: stat of /private/tmp/nix-build-yosys-0.30.drv-0/source/tests/tools/cmp_tbdata failed: No such file or directory
clang-11: error: linker command failed with exit code 134 (use -v to see invocation)
/nix/store/065rv9vqknm20xqrylz7fmmvq6azsagi-post-link-sign-hook: line 4: 57849 Abort trap: 6           CODESIGN_ALLOCATE=codesign_allocate /nix/store/37yhd1z0xw2rpdy5g4k526h1a87i8qm0-sigtool-0.1.3/bin/codesign -f -s - "$linkerOutput"
clang-11: error: linker command failed with exit code 134 (use -v to see invocation)
make[1]: *** [../tools/autotest.mk:10: case_expr_extend.sv] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [../tools/autotest.mk:10: asgn_binop.sv] Error 1
Test: local_loop_var -> ok
Test: case_expr_query -> ok
Test: lesser_size_cast -> ok
Test: implicit_ports -> ok
Test: arrays02 -> ok
Test: defvalue -> ok
make[1]: Leaving directory '/private/tmp/nix-build-yosys-0.30.drv-0/source/tests/simple'
make: *** [Makefile:843: test] Error 2
```

Looks like `doCheck = !(stdenv.isDarwin && stdenv.isAarch64)` should suffice.

--- a/pkgs/development/compilers/yosys/default.nix
+++ b/pkgs/development/compilers/yosys/default.nix
@@ -71,13 +71,13 @@ let
 
 in stdenv.mkDerivation rec {
   pname   = "yosys";
-  version = "0.29";
+  version = "0.30";
 
   src = fetchFromGitHub {
     owner = "YosysHQ";
     repo  = "yosys";
     rev   = "${pname}-${version}";
-    hash  = "sha256-qsuKXYuKsMAALIy1SjxhEhZIMO8B4MF2vmVbwqgzLyM=";
+    hash  = "sha256-qhMcXJFEuBPl7vh+gYTu7PnSWi+L3YMLrBMQyYqfc0w=";
   };
 
   enableParallelBuilding = true;

--- a/pkgs/development/compilers/yosys/plugin-search-dirs.patch
+++ b/pkgs/development/compilers/yosys/plugin-search-dirs.patch
@@ -1,34 +1,45 @@
-diff --git a/passes/cmds/plugin.cc b/passes/cmds/plugin.cc
-index 3ed19497..f9534bd0 100644
---- a/passes/cmds/plugin.cc
-+++ b/passes/cmds/plugin.cc
-@@ -75,8 +75,27 @@ void load_plugin(std::string filename, std::vector<std::string> aliases)
- 		#endif
+diff --git i/passes/cmds/plugin.cc w/passes/cmds/plugin.cc
+index 08b4aa8c4..f00f540e9 100644
+--- i/passes/cmds/plugin.cc
++++ w/passes/cmds/plugin.cc
+@@ -87,15 +87,33 @@ void load_plugin(std::string filename, std::vector<std::string> aliases)
  
- 		void *hdl = dlopen(filename.c_str(), RTLD_LAZY|RTLD_LOCAL);
--		if (hdl == NULL && orig_filename.find('/') == std::string::npos)
--			hdl = dlopen((proc_share_dirname() + "plugins/" + orig_filename + ".so").c_str(), RTLD_LAZY|RTLD_LOCAL);
-+		if (hdl == NULL && orig_filename.find('/') == std::string::npos) {
-+			std::string install_dir = proc_share_dirname() + "plugins";
+ 			// We were unable to open the file, try to do so from the plugin directory
+ 			if (hdl == NULL && orig_filename.find('/') == std::string::npos) {
+-				hdl = dlopen([orig_filename]() {
+-					std::string new_path = proc_share_dirname() + "plugins/" + orig_filename;
++                            std::string install_dir = proc_share_dirname() + "plugins";
+ 
+-					// Check if we need to append .so
+-					if (new_path.find(".so") == std::string::npos)
+-						new_path.append(".so");
++				vector<string> all_dirs;
++				all_dirs.push_back(install_dir);
+ 
+-					return new_path;
+-				}().c_str(), RTLD_LAZY|RTLD_LOCAL);
++				char* plugin_dirs = getenv("NIX_YOSYS_PLUGIN_DIRS");
++				if (plugin_dirs != NULL) {
++					std::string p(plugin_dirs), t;
++					std::stringstream ss(p);
 +
-+			vector<string> all_dirs;
-+			all_dirs.push_back(install_dir);
-+
-+			char* plugin_dirs = getenv("NIX_YOSYS_PLUGIN_DIRS");
-+			if (plugin_dirs != NULL) {
-+				std::string p(plugin_dirs), t;
-+				std::stringstream ss(p);
-+
-+				while(std::getline(ss, t, ':')) {
-+					all_dirs.push_back(t);
++					while(std::getline(ss, t, ':')) {
++						all_dirs.push_back(t);
++					}
 +				}
-+			}
 +
-+			for (auto dir : all_dirs) {
-+				hdl = dlopen((dir + "/" + orig_filename + ".so").c_str(), RTLD_LAZY|RTLD_LOCAL);
-+				if (hdl != NULL) break;
-+			}
-+		}
- 		if (hdl == NULL)
- 			log_cmd_error("Can't load module `%s': %s\n", filename.c_str(), dlerror());
- 		loaded_plugins[orig_filename] = hdl;
++				for (auto dir : all_dirs) {
++					hdl = dlopen([dir, orig_filename]() {
++						std::string new_path = dir + "/" + orig_filename;
++
++						// Check if we need to append .so
++						if (new_path.find(".so") == std::string::npos)
++							new_path.append(".so");
++
++						return new_path;
++					}().c_str(), RTLD_LAZY|RTLD_LOCAL);
++					if (hdl != NULL) break;
++                                }
+ 			}
+ 
+ 			if (hdl == NULL)

